frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton } from '../common';
+
+export default function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceNameValid, setNamespaceNameValid] = useState(false);
+  const [nameHelperMessage, setNameHelperMessage] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    const clusterData = getCluster();
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+        creationTimestamp: new Date().toISOString(),
+        uid: '',
+      },
+    };
+
+    const newNamespaceName = newNamespaceData.metadata.name;
+
+    async function namespaceRequest() {
+      try {
+        const response = await Namespace.apiEndpoint.post(newNamespaceData, {}, clusterData || '');
```

Comment:
If this succeeds we should set the namespace as '', so if we press the button again, we don't see the name of the namespace we just created.